libs/methodologies/bold/rule-processors/mass-id/src/weighing/weighing.helpers.ts

```diff
@@ -111,207 +165,258 @@ export const getWeighingEvents = (document: Document): DocumentEvent[] =>
-): ValidationResult => {
-  const rejectedMessages: string[] = [];
-
-  if (!hasPositiveValue(containerCapacityAttribute)) {
-    rejectedMessages.push(MISSING_RESULT_COMMENTS.CONTAINER_CAPACITY);
-  }
-
-  if (!hasKilogramFormat(containerCapacityAttribute)) {
-    rejectedMessages.push(INVALID_RESULT_COMMENTS.CONTAINER_CAPACITY_FORMAT);
-  }
-
-  return rejectedMessages;
-};
-
-export const validateContainerQuantity = (
-  containerQuantity: MethodologyDocumentEventAttributeValue | undefined,
-  vehicleType: MethodologyDocumentEventAttributeValue | undefined,
-): ValidationResult => {
-  const rejectedMessages: string[] = [];
-
-  if (vehicleType === TRUCK && isNonZeroPositiveInt(containerQuantity)) {
-    rejectedMessages.push(INVALID_RESULT_COMMENTS.CONTAINER_QUANTITY);
-  }
-
-  if (!isNonZeroPositiveInt(containerQuantity)) {
-    rejectedMessages.push(MISSING_RESULT_COMMENTS.CONTAINER_QUANTITY);
-  }
-
-  return rejectedMessages;
-};
-
-export const validateGrossWeightAttribute = (
-  grossWeight: MethodologyDocumentEventAttribute | undefined,
-): ValidationResult => {
-  const rejectedMessages: string[] = [];
-
-  if (!hasPositiveValue(grossWeight)) {
-    rejectedMessages.push(
-      MISSING_RESULT_COMMENTS.GROSS_WEIGHT(grossWeight?.value),
-    );
-  }
-
-  if (!hasKilogramFormat(grossWeight)) {
-    rejectedMessages.push(INVALID_RESULT_COMMENTS.GROSS_WEIGHT_FORMAT);
-  }
-
-  return rejectedMessages;
-};
-
-export const validateMassNetWeightAttribute = (
-  massNetWeightAttribute: MethodologyDocumentEventAttribute | undefined,
-): ValidationResult => {
-  const rejectedMessages: string[] = [];
-
-  if (!hasPositiveValue(massNetWeightAttribute)) {
-    rejectedMessages.push(
-      MISSING_RESULT_COMMENTS.MASS_NET_WEIGHT(massNetWeightAttribute?.value),
-    );
-  }
-
-  if (!hasKilogramFormat(massNetWeightAttribute)) {
-    rejectedMessages.push(INVALID_RESULT_COMMENTS.MASS_NET_WEIGHT_FORMAT);
-  }
+type Validator = (values: WeighingValues, isTwoStep?: boolean) => string[];
+
+const validators: Record<string, Validator> = {
+  containerCapacity: (values) => {
+    if (!isNil(values.containerCapacityException)) {
+      return [];
+    }
+
+    const messages: string[] = [];
+
+    if (!hasPositiveValue(values.containerCapacityAttribute)) {
+      messages.push(MISSING_RESULT_COMMENTS.CONTAINER_CAPACITY);
+    }
+
+    if (!hasKilogramFormat(values.containerCapacityAttribute)) {
+      messages.push(INVALID_RESULT_COMMENTS.CONTAINER_CAPACITY_FORMAT);
+    }
+
+    return messages;
+  },
+
+  containerQuantity: (values) => {
+    const messages: string[] = [];
+
+    if (
+      values.vehicleType === TRUCK &&
+      isNonZeroPositiveInt(values.containerQuantity)
+    ) {
+      messages.push(INVALID_RESULT_COMMENTS.CONTAINER_QUANTITY);
+    }
+
+    if (!isNonZeroPositiveInt(values.containerQuantity)) {
+      messages.push(MISSING_RESULT_COMMENTS.CONTAINER_QUANTITY);
```

Comment:
I think the constant name `MISSING_RESULT_COMMENTS` is misleading in this and similar cases: we are not handling a missing case, we are checking its format, so it would be better with something like `WRONG_FORMAT_RESULT_COMMENTS`. Even if it is missing, it is not in the correct format, but the other way around is not true.